runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowPipelineJob.java

```diff
@@ -250,7 +249,12 @@ State waitUntilFinish(
       }
 
       if (!hasError) {
+        // Reset the backoff, and update the new cumulative sleep time to be the remaining
+        // part of the total input sleep duration.
         backoff.reset();
+        long nanosConsumed = nanoClock.nanoTime() - startNanos;
+        Duration consumed = Duration.millis((nanosConsumed + 999999) / 1000000);
+        backoff = backoff.withMaxCumulativeBackoff(duration.minus(consumed));
```

Comment:
I thought about then when replacing the other class, but I think that it is a bad break of the BackOff abstraction. For example:

- `reset()` only doing a partial reset (attempts, not time)
- definition of start time (is it when the BackOff is constructed? when the first call to `nextBackOffMillis()`? Something else?

It seemed unnecessary complexity for the `BackOff` object itself. Callers should be the ones who care about this and worry about the details. It's not that bad and it keeps the purpose of `BackOff` as it is documented today.